rpc/api/eth.go

```diff
@@ -578,14 +579,15 @@ func (self *ethApi) Resend(req *shared.Request) (interface{}, error) {
 		return nil, shared.NewDecodeParamError(err.Error())
 	}
 
-	ret, err := self.xeth.Transact(args.Tx.From, args.Tx.To, args.Tx.Nonce, args.Tx.Value, args.GasLimit, args.GasPrice, args.Tx.Data)
-	if err != nil {
-		return nil, err
+	pending := self.ethereum.TxPool().GetTransactions()
+	for _, p := range pending {
+		if found := p.SigHash() == args.Tx.tx.SigHash(); found {
```

Comment:
`SigHash` is expensive. Maybe that doesn't matter in this case, just informing you about it.
Might be better to compare `From` and `Nonce`. They uniquely identify the transaction for the purpose of resend.